core/envelope-sim/src/test/java/fr/sncf/osrd/envelope_sim/AllowanceRangesTests.java

```diff
@@ -320,17 +326,19 @@ public void regressionTestCornerCase() {
                 )
         );
         var maxEffortEnvelope = makeSimpleMaxEffortEnvelope(testContext, 80, stops);
-        var err = assertThrows(AllowanceConvergenceException.class,
-                () -> allowance.apply(maxEffortEnvelope, testContext));
-        assertEquals(AllowanceConvergenceException.ErrorType.TOO_MUCH_TIME, err.errorType);
+        var err = assertThrows(
+                OSRDError.class,
+                () -> allowance.apply(maxEffortEnvelope, testContext)
+        );
+        assertEquals(err.osrdErrorType, ErrorType.AllowanceConvergenceTooMuchTime);
     }
 
     /** Applies the allowance to the envelope. Any user error (impossible margin) is ignored */
     private void applyAllowanceIgnoringUserError(Allowance allowance, Envelope envelope, EnvelopeSimContext context) {
         try {
             allowance.apply(envelope, context);
-        } catch (AllowanceConvergenceException e) {
-            assertEquals(OSRDError.ErrorCause.USER, e.cause);
+        } catch (OSRDError e) {
+            assertEquals(e.cause, ErrorCause.USER);
```

Comment:
This catches more errors than before, it should only ignore the allowance related errors

Reply:
not exactly sure how to do that

Reply:
made the change, 
```
            if (
                    e.osrdErrorType == ErrorType.AllowanceConvergenceDiscontinuity
                    || e.osrdErrorType == ErrorType.AllowanceConvergenceNotEnoughTime
                    || e.osrdErrorType == ErrorType.AllowanceConvergenceTooMuchTime
            ) {
                assertEquals(e.cause, ErrorCause.USER);
            }
```